### PR TITLE
[RW-7121][risk=no] Surface runtime error messages to users via UI

### PR DESCRIPTION
**Pull request by als364**
Screen Shot 2021-09-21 at 11 30 23 AM
Screen Shot 2021-09-21 at 2 38 00 PM

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/analysis/runtime-panel.tsx [hunk 1]
```diff
@@ -1594,7 +1617,11 @@ const RuntimePanel = fp.flow(
                   aria-label='Delete Environment'
                   disabled={disableControls || !runtimeExists}
                   onClick={() => setPanelContent(PanelContent.DeleteRuntime)}>Delete Environment</Link>
-              {runtimeExists || (pdExists && pdSizeReduced) ? renderNextButton() : renderCreateButton()}
+              {runtimeExists || (pdExists && pdSizeReduced)
+                  ? renderNextButton()
+                  : currentRuntime && currentRuntime.errors
```
Comment (calbach) [inline ui/src/app/pages/analysis/runtime-panel.tsx:1622]: nit: could you use switchCase here instead? The double ternary hurts my eyes

Comment (jmthibault79) [reply ui/src/app/pages/analysis/runtime-panel.tsx:1622]: I like utils/index `cond()` in a case like this where it's a little more complex than a switch(constant)

File: ui/src/app/components/help-sidebar.tsx [hunk 2]
```diff
@@ -419,6 +425,22 @@ export const HelpSidebar = fp.flow(
...
+        // If the runtime status has changed from something that is not Error to Error,
+        // and we have error messages, show the error modal.
+        if (
+            newRuntime.runtimeLoaded
+            && newRuntime.workspaceNamespace === this.props.workspace.namespace
+            && newRuntime.runtime.status === RuntimeStatus.Error
+            && (!oldRuntime.runtime || oldRuntime.runtime.status !== RuntimeStatus.Error)
+            && newRuntime.runtime.errors
+        ) {
+          this.setState({
```
[the remark this answers is not in the document]
Comment (als364) [reply ui/src/app/components/help-sidebar.tsx:438]: We do need the explicit state tracking - this callback fires before the runtime prop updates.

File: ui/src/app/pages/analysis/runtime-panel.tsx (hunk 1 above)
Comment (als364) [reply ui/src/app/pages/analysis/runtime-panel.tsx:1622]: Ooh, `cond` is a good idea

File: ui/src/app/components/runtime-error-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          type='secondary'
+          onClick={() => closeFunction()}
+      >
+        Close
+      </Button>
+      <Button onClick={() => {
+        openRuntimePanel();
+        closeFunction();
+      }}>
+        Open runtime panel
```
[the remark this answers is not in the document]
Comment (als364) [reply ui/src/app/components/runtime-error-modal.tsx:29]: and even that would be pretty big but it still fits in a normal size modal

**Comment by als364** [comment]
Puppeteer failure is due to RW-7302; merging

**Pull request merged**